src/serialization.rs

```diff
@@ -0,0 +1,171 @@
+use bitcoin::consensus::encode::{Decodable, Encodable, ReadExt, WriteExt};
+use bitcoin::secp256k1::{SecretKey, Signature};
+use bitcoin::{PublicKey, Script, VarInt};
+
+// Maximum size, in bytes, of a vector we are allowed to decode
+// TODO: Decide weather its useful
+pub const MAX_VEC_SIZE: usize = 4_000_000;
+
+#[derive(Debug)]
+pub enum SerialisationError {
+    IO(std::io::Error),
+
+    ConsensusEcode(bitcoin::consensus::encode::Error),
+
+    Secp256k1(bitcoin::secp256k1::Error),
+
+    Key(bitcoin::util::key::Error),
+
+    OverSizedAllocation { requested: usize, max: usize },
+
+    General(&'static str),
+}
+
+impl From<bitcoin::secp256k1::Error> for SerialisationError {
+    fn from(e: bitcoin::secp256k1::Error) -> Self {
+        SerialisationError::Secp256k1(e)
+    }
+}
+
+impl From<bitcoin::consensus::encode::Error> for SerialisationError {
+    fn from(e: bitcoin::consensus::encode::Error) -> Self {
+        SerialisationError::ConsensusEcode(e)
+    }
+}
+
+impl From<std::io::Error> for SerialisationError {
+    fn from(e: std::io::Error) -> Self {
+        SerialisationError::IO(e)
+    }
+}
+
+impl From<bitcoin::util::key::Error> for SerialisationError {
+    fn from(e: bitcoin::util::key::Error) -> Self {
+        SerialisationError::Key(e)
+    }
+}
+
+// A Serializable Trait that defines byte encoding of structures
+// Anywhere we need byte encoded data (Network, Database) we can use this.
+// The underlying encoding of primitve datatyes and structures are done
+// via bitcoin::consensus::encode::Encodale trait.
+pub trait Serializable {
+    fn serialise<W: std::io::Write>(&self, w: W) -> Result<usize, SerialisationError>;
```

Comment:
Maybe instead of the intentional typo here, below, and in the error type, you can use the normal spelling. When there is a naming import conflict with `serde::Serializable` you can import the local function as `TeleportSerializable` or keep the mod path like `serialization::Serializable`.

Small typos like this, even if intentional, make the code harder to read, IMHO. The extra mental load of "does this come from our local serialization code, and I need to type serialise" also seems like it would get annoying pretty quickly.